### ISSUE TITLE
ReduceLROnPlateau callback from Tf keras is not recognizing cohen kappa metrics direction in 'Auto' mode

### ISSUE THREAD

**Issue by yuvaramsingh94** [tensorflow/addons#1865]
Hi all,
    when i use ReduceLROnPlateau callback along with tfa cohen kappa, the 'Auto' mode of ReduceLROnPlateau is not able to detect the direction (increase or decrease). the mode should be specifically set to 'max'. can this be solved so that the Auto mode can monitor the metric and take care of Lr reduction.

**Comment by bhack** [comment]
> in auto mode, the direction is automatically inferred from the name of the monitored quantity

**Comment by yuvaramsingh94** [comment]
@bhack thanks. but, it is not automatically inferring it . this same issue is happening in ModelCheckpoint callback also when i set the mode auto it did not save the best weight and it just saved some weight. 

in the ReduceLROnPlateau  case, the callback automatically kicked in eventhough i can see the cohen kappa metric value is going up .

**Comment by bhack** [comment]
`min` and `auto` (if not in the monitor list) enter in the same condition https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/python/keras/callbacks.py#L2229

**Comment by yuvaramsingh94** [comment]
`if (self.mode == 'min' or (self.mode == 'auto' and 'acc' not in self.monitor)):` is this the problem?

as we are going to set `monitor = val_cohen_kappa` which is not having `acc` string so it is not automatically recognizing the direction.

something with ModelCheckpoint as i lost like 2 days trying to train a model and i ended up with a poor model weight saved.

i think there should be a method to atleast know the direction of a metrics that are from tfa so it can avoid such scenarios.

Anyways , thanks @bhack

**Comment by bhack** [comment]
Yes the API clearly tell you that is based on the name. 
Probably in Tensorflow docs it could be more explict about the specific sub-string It Is searching for in the name.

I think the best thing to do Is to open a Doc type issue in the Tensorflow repo directly.

**Comment by yuvaramsingh94** [comment]
got it . will do that

**Comment by bhack** [comment]
If you are still interested you need to open a doc PR on the [Keras repository](https://github.com/keras-team/keras).
